applications/CompressiblePotentialFlowApplication/custom_elements/transonic_perturbation_potential_flow_element.cpp

```diff
@@ -379,6 +411,21 @@ void TransonicPerturbationPotentialFlowElement<TDim, TNumNodes>::GetWakeDistance
     noalias(distances) = GetValue(WAKE_ELEMENTAL_DISTANCES);
 }
 
+template <int TDim, int TNumNodes>
+void TransonicPerturbationPotentialFlowElement<TDim, TNumNodes>::GetEquationIdVectorExtendedElement(
+    EquationIdVectorType& rResult) const
+{
+    const int additional_node_id = GetAdditionalNode();
+
+    const auto& r_geometry = this->GetGeometry();
+    for (unsigned int i = 0; i < TNumNodes; i++)
+    {
+        rResult[i] = r_geometry[i].GetDof(VELOCITY_POTENTIAL).EquationId();
+    }
+    const auto& r_upwind_geometry = mpUpwindElement->GetGeometry();
+    rResult[TNumNodes + 1] = r_upwind_geometry[additional_node_id].GetDof(VELOCITY_POTENTIAL).EquationId();
+}
+
```

Comment:
I like the algorithm and the fact that you put it in a separate function called GetEquationIdVectorExtendedElement. Here is an extension based in your implementation in order to treat both current and upwind kutta elements:

```suggestion
template <int TDim, int TNumNodes>
void TransonicPerturbationPotentialFlowElement<TDim, TNumNodes>::GetEquationIdVectorExtendedElement(
    EquationIdVectorType& rResult) const
{
    const TransonicPerturbationPotentialFlowElement& r_this = *this;
    const int kutta = r_this.GetValue(KUTTA);
    if(kutta == 0){
        GetEquationIdVectorNormalElement(rResult);
    }
    else{
        GetEquationIdVectorKuttaElement(rResult);
    }

    const int additional_upwind_node_index = GetAdditionalUpwindNodeIndex();
    const auto& r_upstream_element = *pGetUpwindElement();
    const auto& r_upwind_geometry = r_upstream_element.GetGeometry();
    const int upstream_kutta = r_upstream_element.GetValue(KUTTA);
    if (upstream_kutta == 0) { // upwind element is not kutta
        //TODO Special treatment for the upwind wake elements
        rResult[TNumNodes] = r_upwind_geometry[additional_upwind_node_index]
                                     .GetDof(VELOCITY_POTENTIAL)
                                     .EquationId();
    }
    else { // upwind element is kutta
        if (!r_upwind_geometry[additional_upwind_node_index].GetValue(TRAILING_EDGE)) {
            // upwind node is not trailing edge
            rResult[TNumNodes] = r_upwind_geometry[additional_upwind_node_index]
                                         .GetDof(VELOCITY_POTENTIAL)
                                         .EquationId();
        } else { // upwind node is trailing edge
            rResult[TNumNodes] = r_upwind_geometry[additional_upwind_node_index]
                                         .GetDof(AUXILIARY_VELOCITY_POTENTIAL)
                                         .EquationId();
        }
    }
}
```